### ISSUE TITLE
step alias should not allow thing containing a .

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-tool_dataflows#372]
eg `send_mdl_quiz.json_to_s3`

This could easily be ambogous if there was another step with an alias of `send_mdl_quiz`

**Comment by jaypha** [comment]
Should 'name' have a similar constraint?

**Comment by jaypha** [comment]
I think it should also check for spaces.

**Comment by brendanheywood** [comment]
Name is supposed to be human readable so anything goes there. alias I'd like to be really simple perhaps PARAM_ALPHANUMTEXT, letters numbers and _

**Comment by keevan** [comment]
Yes, there should be validation to ensure the alias is in the expected format which won't cause issues. During my testing I found the best format (which is also widely used) to be snake casing, which is alphanumeric, uses underscores which work well for dot access.

The current bit that converts the name is here: https://github.com/catalyst/moodle-tool_dataflows/blob/MOODLE_35_STABLE/classes/step.php#L267 .  I've added in strtolower (lowercase) for consistency with regards to usage, but this isn't a requirement.

The above code should be moved out to a helper function which does the conversion for us, and turns whatever input into a valid formatted alias.
